cli/src/pcluster/constants.py

```diff
@@ -337,3 +337,5 @@ class Operation(Enum):
 IAM_ROLE_REGEX = "^arn:.*:role/"
 IAM_INSTANCE_PROFILE_REGEX = "^arn:.*:instance-profile/"
 IAM_POLICY_REGEX = "^arn:.*:policy/"
+
+CFN_BOOTSTRAP_VIRTUALENV_PATH = "/opt/parallelcluster/pyenv/versions/3.9.17/envs/cfn_bootstrap_virtualenv/bin"
```

Comment:
I don't like the fact we're adding in the cli the relation with the virtualenv created by the cookbook and with python version used by the cookbook too.

What we can do is to continue to the profile.sh file, renaming it to `cookbook_environment.sh` or something like this and rather than setting the `PATH`  into it, define an `export CFN_BOOTSTRAP_VIRTUALENV_PATH=something` on it.

So in the cli we need to source the `cookbook_environment.sh` and then simply use the `CFN_BOOTSTRAP_VIRTUALENV_PATH` that will be exported in the environment.

In this way the cookbook can change the path, the python version etc in a safe way, without impacting the cli code.

Reply:
I like this suggestion, will update and test accordingly